### PR TITLE
docs: add note about PYTHONMALLOC for accurate jemalloc memory tracking

### PR DESCRIPTION
**Pull request by hensg**
Added a note in the documentation suggesting that users may set `PYTHONMALLOC=malloc` when using `jemalloc`. This allows jemalloc to track memory usage more accurately by bypassing Python's internal small-object allocator (`pymalloc`), helping to ensure that `cache_autotuning` functions as expected.

This doc change aims to provide more clarity for users configuring jemalloc with Synapse.

Based on: https://github.com/element-hq/synapse/blob/4ac783549c5bac7a490a715d359f330bb0b1a161/synapse/metrics/jemalloc.py#L198-L201

### CODE REVIEW

**Comment by daedric7** [comment]
> Added a note in the documentation suggesting that users may set `PYTHONMALLOC=malloc` when using `jemalloc`. This allows jemalloc to track memory usage more accurately by bypassing Python's internal small-object allocator (`pymalloc`), helping to ensure that `cache_autotuning` functions as expected.
> 
> This doc change aims to provide more clarity for users configuring jemalloc with Synapse.

I tried this setting with Docker Compose. Memory usage did go down, yet became much more erratic. On the following graph, the switching point is around 14h00.

image

**Comment by hensg** [comment]
@daedric7 could you share the version without malloc/jemalloc and your configurations?

**Comment by hensg** [comment]
Are you using `jemalloc` @daedric7 ? e.g.: `LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libjemalloc.so.2`

**Comment by daedric7** [comment]
> Are you using `jemalloc` @daedric7 ? e.g.: `LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libjemalloc.so.2`

As far as i was aware, jemalloc2 it's always and use when using Docker. 

Is this not true ?

**Comment by daedric7** [comment]
> > Are you using `jemalloc` @daedric7 ? e.g.: `LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libjemalloc.so.2`
> 
> As far as i was aware, jemalloc2 it's always and use when using Docker.
> 
> Is this not true ?

I checked start.py inside my container:

```
   jemallocpath = "/usr/lib/%s-linux-gnu/libjemalloc.so.2" % (platform.machine(),)

    if os.path.isfile(jemallocpath):
        environ["LD_PRELOAD"] = jemallocpath
    else:
        log("Could not find %s, will not use" % (jemallocpath,))

```

Can you please confirm that jemalloc2 is not used by default with docker ? Everything  else points to the oposite:

```
###
### Stage 2: runtime
###

FROM docker.io/library/python:${PYTHON_VERSION}-slim-bookworm

LABEL org.opencontainers.image.url='https://matrix.org/docs/projects/server/synapse'
LABEL org.opencontainers.image.documentation='https://github.com/element-hq/synapse/blob/master/docker/README.md'
LABEL org.opencontainers.image.source='https://github.com/element-hq/synapse.git'
LABEL org.opencontainers.image.licenses='AGPL-3.0-or-later'

RUN \
  --mount=type=cache,target=/var/cache/apt,sharing=locked \
  --mount=type=cache,target=/var/lib/apt,sharing=locked \
  apt-get update -qq && apt-get install -yqq \
  curl \
  gosu \
  libjpeg62-turbo \
  libpq5 \
  libwebp7 \
  xmlsec1 \
  libjemalloc2 \
  libicu72 \
  libssl-dev \
  openssl \
  && rm -rf /var/lib/apt/lists/* 
```

File: changelog.d/17709.doc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Add documentation note about PYTHONMALLOC for accurate jemalloc memory tracking
+Contributed by @hensg.
```
Comment (erikjohnston) [inline changelog.d/17709.doc:2]:
```suggestion
Add documentation note about PYTHONMALLOC for accurate jemalloc memory tracking. Contributed by @hensg.
```

**Review by erikjohnston — APPROVED** [review]
Thanks!

File: changelog.d/17709.doc (hunk 1 above)
Comment (erikjohnston) [reply changelog.d/17709.doc:2]: (All our changelog entries are one line)

**Pull request merged**